### ISSUE TITLE
Mac OS editor build failing

### ISSUE THREAD

**Issue by matthewhilton** [Zylann/godot_voxel#526]
I'm following https://voxel-tools.readthedocs.io/en/latest/getting_the_module/ to build a MacOS Godot editor, but getting errors such as 

```
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:3:10: error: '../../../util/godot/classes/array_mesh.h' file not found, did you mean 'util/godot/classes/array_mesh.h'?
```

I'm building the latest master Godot  [030c1a950e8cee19e72104b4fb79caba17a2d678](https://github.com/godotengine/godot/commit/3710f069293f2fde8afe33fea898c4b36fa5e943) and the voxel tools current master c4ebc53ec64bfc3c63f5096af8e39235f9404114

I was able to get it to build when using an older version of voxel tools: 6064a4c156992344bdcce7654c1d8ebd8b80b5e2 (I just chose a random older version, a new version than this may build but I have not checked)

So I assume there is a change between 6064a4c156992344bdcce7654c1d8ebd8b80b5e2 and c4ebc53ec64bfc3c63f5096af8e39235f9404114 that is breaking the build, or there is a change in the locations and they just need to be updated

Similar error in https://github.com/Zylann/godot_voxel/issues/501

Full error:

```
[ 77%] Compiling modules/voxel/editor/blocky_library/model_viewer.cpp ...
modules/voxel/editor/blocky_library/axes_3d_control.cpp:2:10: error: '../../../util/math/funcs.h' file not found, did you mean 'util/math/funcs.h'?
#include "../../../util/math/funcs.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/math/funcs.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:2:10: error: '../../../constants/voxel_string_names.h' file not found, did you mean 'constants/voxel_string_names.h'?
#include "../../../constants/voxel_string_names.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "constants/voxel_string_names.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:3:10: error: '../../../util/godot/classes/array_mesh.h' file not found, did you mean 'util/godot/classes/array_mesh.h'?
#include "../../../util/godot/classes/array_mesh.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/array_mesh.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:4:10: error: '../../../util/godot/classes/button.h' file not found, did you mean 'util/godot/classes/button.h'?
#include "../../../util/godot/classes/button.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/button.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:5:10: error: '../../../util/godot/classes/editor_undo_redo_manager.h' file not found, did you mean 'util/godot/classes/editor_undo_redo_manager.h'?
#include "../../../util/godot/classes/editor_undo_redo_manager.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/editor_undo_redo_manager.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:6:10: error: '../../../util/godot/classes/mesh_instance_3d.h' file not found, did you mean 'util/godot/classes/mesh_instance_3d.h'?
#include "../../../util/godot/classes/mesh_instance_3d.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/mesh_instance_3d.h"
2 errors generated.
scons: *** [modules/voxel/editor/blocky_library/axes_3d_control.macos.editor.arm64.o] Error 1
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:7:10: error: '../../../util/godot/classes/scene_tree.h' file not found, did you mean 'util/godot/classes/scene_tree.h'?
#include "../../../util/godot/classes/scene_tree.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/scene_tree.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:8:10: error: '../../../util/godot/classes/standard_material_3d.h' file not found, did you mean 'util/godot/classes/standard_material_3d.h'?
#include "../../../util/godot/classes/standard_material_3d.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/standard_material_3d.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:9:10: error: '../../../util/godot/classes/v_box_container.h' file not found, did you mean 'util/godot/classes/v_box_container.h'?
#include "../../../util/godot/classes/v_box_container.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/v_box_container.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:10:10: error: '../../../util/godot/core/array.h' file not found, did you mean 'util/godot/core/array.h'?
#include "../../../util/godot/core/array.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/core/array.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:11:10: error: '../../../util/godot/core/callable.h' file not found, did you mean 'util/godot/core/callable.h'?
#include "../../../util/godot/core/callable.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/core/callable.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:12:10: error: '../../../util/godot/core/mouse_button.h' file not found, did you mean 'util/godot/core/mouse_button.h'?
#include "../../../util/godot/core/mouse_button.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/core/mouse_button.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:13:10: error: '../../../util/godot/core/string.h' file not found, did you mean 'util/godot/core/string.h'?
#include "../../../util/godot/core/string.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/core/string.h"
modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.cpp:14:10: error: '../../../util/godot/editor_scale.h' file not found, did you mean 'util/godot/editor_scale.h'?
#include "../../../util/godot/editor_scale.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/editor_scale.h"
13 errors generated.
scons: *** [modules/voxel/editor/blocky_library/voxel_blocky_model_viewer.macos.editor.arm64.o] Error 1
modules/voxel/editor/blocky_library/model_viewer.cpp:2:10: error: '../../../util/godot/classes/camera_3d.h' file not found, did you mean 'util/godot/classes/camera_3d.h'?
#include "../../../util/godot/classes/camera_3d.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/camera_3d.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:3:10: error: '../../../util/godot/classes/directional_light_3d.h' file not found, did you mean 'util/godot/classes/directional_light_3d.h'?
#include "../../../util/godot/classes/directional_light_3d.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/directional_light_3d.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:4:10: error: '../../../util/godot/classes/input_event_mouse_motion.h' file not found, did you mean 'util/godot/classes/input_event_mouse_motion.h'?
#include "../../../util/godot/classes/input_event_mouse_motion.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/input_event_mouse_motion.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:5:10: error: '../../../util/godot/classes/sub_viewport.h' file not found, did you mean 'util/godot/classes/sub_viewport.h'?
#include "../../../util/godot/classes/sub_viewport.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/sub_viewport.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:6:10: error: '../../../util/godot/classes/sub_viewport_container.h' file not found, did you mean 'util/godot/classes/sub_viewport_container.h'?
#include "../../../util/godot/classes/sub_viewport_container.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/sub_viewport_container.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:7:10: error: '../../../util/godot/classes/world_3d.h' file not found, did you mean 'util/godot/classes/world_3d.h'?
#include "../../../util/godot/classes/world_3d.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/world_3d.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:8:10: error: '../../../util/godot/core/mouse_button.h' file not found, did you mean 'util/godot/core/mouse_button.h'?
#include "../../../util/godot/core/mouse_button.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/core/mouse_button.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:9:10: error: '../../../util/godot/editor_scale.h' file not found, did you mean 'util/godot/editor_scale.h'?
#include "../../../util/godot/editor_scale.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/editor_scale.h"
modules/voxel/editor/blocky_library/model_viewer.cpp:10:10: error: '../../../util/math/funcs.h' file not found, did you mean 'util/math/funcs.h'?
#include "../../../util/math/funcs.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/math/funcs.h"
In file included from modules/voxel/editor/blocky_library/model_viewer.cpp:11:
modules/voxel/editor/blocky_library/axes_3d_control.h:4:10: error: '../../../util/godot/classes/control.h' file not found, did you mean 'util/godot/classes/control.h'?
#include "../../../util/godot/classes/control.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         "util/godot/classes/control.h"
10 errors generated.
scons: *** [modules/voxel/editor/blocky_library/model_viewer.macos.editor.arm64.o] Error 1
scons: building terminated because of errors.
```

**Comment by Zylann** [comment]
In #501 the issue was that Godot devs moved a header file somewhere else in Godot 4.1, so the module had to conditionally include either the old location from 4.0 or the new location from 4.1.

In your error log there doesn't seem to be files coming from Godot though.

Looking at `axes_3d_control.cpp`, I notice that there is an extra `../` in `#include "../../../util/math/funcs.h"`. It should not be present.
I don't understand why GCC and MSVC builds on CI never had a problem with this, they found the file somehow even thought the path was wrong Oo In VSCode intellisense doesn't even complain and the code compiles, wether I remove the extra `../` or not. But it complains if I remove the line entirely 🤷. There is clearly an issue, but these weird behaviors are not helping to find it^^"

Submitted a fix in 2720e1988b5b754eae5b93d0f9c4270da70dcb80

**Comment by matthewhilton** [comment]
Yes I was a bit stumped as well why CI builds but locally doesn't 🤷 

Regardless - https://github.com/Zylann/godot_voxel/commit/2720e1988b5b754eae5b93d0f9c4270da70dcb80 builds successfully

Thanks for the prompt response!
